addons/chopping/scripts/Game/ACE_Chopping/UserActions/ACE_Chopping_UserAction.c

```diff
@@ -2,30 +2,46 @@
 //! Tree deletion user action
 class ACE_Chopping_UserAction : ACE_ShovelUserAction
 {
-    //------------------------------------------------------------------------------------------------
-    //! Request deletion of the tree
-    override void PerformAction(IEntity pOwnerEntity, IEntity pUserEntity) 
-    {
-        SCR_PlayerController userCtrl = SCR_PlayerController.Cast(GetGame().GetPlayerController());
-        if (!userCtrl)
-            return;
-        
-        ACE_Chopping_HelperEntity helper = ACE_Chopping_HelperEntity.Cast(GetOwner());
-        if (!helper)
-            return;
-        
-        IEntity plant = helper.GetAssociatedPlant();
-        if (!plant)
-            return;
-        
-		userCtrl.ACE_RequestDeleteEntity(plant);
+	protected static const int DELETE_FALLING_TREE_DELAY_MS = 10000;
+	
+	//------------------------------------------------------------------------------------------------
+	//! Request deletion of the tree
+	override void PerformAction(IEntity pOwnerEntity, IEntity pUserEntity)
+	{
+		SCR_PlayerController userCtrl = SCR_PlayerController.Cast(GetGame().GetPlayerController());
+		if (!userCtrl)
+			return;
+		
+		ACE_Chopping_HelperEntity helper = ACE_Chopping_HelperEntity.Cast(GetOwner());
+		if (!helper)
+			return;
+		
+		Tree plant = Tree.Cast(helper.GetAssociatedPlant());
+		if (!plant)
+			return;
+		
+		bool enabled;
+		BaseContainer container = plant.GetPrefabData().GetPrefab();
+		
+		if (container && container.Get("Enabled", enabled) && enabled)
+		{
+			vector transform[4];
+			pUserEntity.GetWorldTransform(transform);
+			userCtrl.ACE_RequestDestroyEntity(plant, EDamageType.MELEE, transform);
+			GetGame().GetCallqueue().CallLater(userCtrl.ACE_RequestDeleteEntity, DELETE_FALLING_TREE_DELAY_MS, false, plant);
```

Comment:
One thing to consider is that the tree won't be deleted if client disconnects before this happens. Not a big issue.

Reply:
Good point, I should make a request to the server instead.